smm/lotus_adapter.go

```diff
@@ -0,0 +1,253 @@
+    if err != nil {
+        return nil, err
+    }
+    state.Sectors = make(map[uint64]api.ChainSectorInfo)
+    for _, sectorInfo := range sectors {
+        state.Sectors[sectorInfo.SectorID] = *sectorInfo
+    }
+    for _, sectorInfo := range provingSectors {
+        state.ProvingSet[sectorInfo.SectorID] = struct{}{}
+    }
+    state.Address = Address(adapter.miner.String())
+    state.PreCommittedSectors = make(map[uint64]api.ChainSectorInfo)
+    state.StagedCommittedSectors = make(map[uint64]api.ChainSectorInfo)
+    state.ProvingSet = make(map[uint64]struct{})
+    return state, nil
+}
+
+func (adapter lotusAdapter) GetRandomness(ctx context.Context, stateKey StateKey, offset uint) ([]byte, error) {
+    ts, err := statekey2tipset(ctx, stateKey, adapter.fullNode)
+    if err != nil {
+        return nil, err
+    }
+    tsk := types.NewTipSetKey(ts.Cids()...)
+    return adapter.fullNode.ChainGetRandomness(ctx, tsk, ts.Blocks()[0].Tickets, int(offset))
+}
+
+func (adapter lotusAdapter) GetProvingPeriod(ctx context.Context, stateKey StateKey) (*ProvingPeriod, error) {
+    pp := new(ProvingPeriod)
+    ts, err := statekey2tipset(ctx, stateKey, adapter.fullNode)
+    if err != nil {
+        return nil, err
+    }
+    end, err := adapter.fullNode.StateMinerProvingPeriodEnd(ctx, adapter.miner, ts)
+    if err != nil {
+        return nil, err
+    }
+    pp.End = Epoch(end)
+    if end > build.ProvingPeriodDuration {
+        pp.Start = Epoch(end - build.ProvingPeriodDuration)
+    } else {
+        pp.Start = 0
+    }
+
+    return pp, nil
+}
+
+func (adapter lotusAdapter) SubmitSelfDeals(ctx context.Context, deals []uint64) (cid.Cid, error) {
+    // TODO: Adapt from shared modules' definition of storage deals to lotus' storage deal
+    // TODO: Each deal has to be signed (adapter.fullNode.WalletSign)
+    params := actors.PublishStorageDealsParams{
+        Deals: make([]actors.StorageDeal, len(deals)),
+    }
+    return adapter.callMinerActorMethod(ctx, actors.SMAMethods.PublishStorageDeals, &params)
+}
+
+func (adapter lotusAdapter) SubmitSectorPreCommitment(ctx context.Context, id SectorID, commR cid.Cid, dealIDs []uint64) (cid.Cid, error) {
+    params := actors.SectorPreCommitInfo{
+        SectorNumber: uint64(id),
+        CommR: commR.Bytes(),
+        SealEpoch: 0, // TODO: does this need to be passed in?!
+        DealIDs: dealIDs,
+    }
+    return adapter.callMinerActorMethod(ctx, actors.MAMethods.PreCommitSector, &params)
+}
+
+func (adapter lotusAdapter) GetSealSeed(ctx context.Context, state StateKey, id SectorID) SealSeed {
+    // TODO: Not sure how to implement this
+    panic("implement me")
+}
+
+func (adapter lotusAdapter) SubmitSectorCommitment(ctx context.Context, id SectorID, proof Proof, dealIDs []uint64) (cid.Cid, error) {
+    params := actors.SectorProveCommitInfo{
+        Proof: proof,
+        SectorID: uint64(id),
+        DealIDs: dealIDs,
```

Comment:
Deal ids shouldn't be required here, only for pre-commitment. If this is a requirement of the upstream, leave a comment here noting that it should change.